ipatests/test_integration/test_installation.py

```diff
@@ -316,3 +316,42 @@ def test_install_master(self):
 
     def test_install_kra(self):
         tasks.install_kra(self.master, first_instance=True)
+
+
+class TestInstallMasterReservedIPasForwarder(IntegrationTest):
+    """IANA reserved IP address can not be used as a forwarder.
+    This test checks if ipa server installation throws an error when
+    0.0.0.0 is specified as forwarder IP address.
+
+    related ticket: https://pagure.io/freeipa/issue/6894 """
+
+    def test_reserved_ip_as_forwarder(self):
+        args = [
+            'ipa-server-install',
+            '-n', self.master.domain.name,
+            '-r', self.master.domain.realm,
+            '-p', self.master.config.dirman_password,
+            '-a', self.master.config.admin_password,
+            '--setup-dns',
+            '--forwarder', '0.0.0.0',
+            '--auto-reverse']
+        cmd = self.master.run_command(args, raiseonerr=False)
+        assert cmd.returncode == 2
+        exp_str = ("error: option --forwarder: invalid IP address 0.0.0.0: "
+                   "cannot use IANA reserved IP address 0.0.0.0")
+        assert exp_str in cmd.stderr_text
+
+        dirman_password = self.master.config.dirman_password
+        admin_password = self.master.config.admin_password
+        ip = "0.0.0.0"
+        server_install_options = ("yes\n\n\n\n"
+            "{dirman_password}\n{dirman_password}"
+            "\n{admin_password}\n{admin_password}\n"
+            "yes\nyes\n{ip}\n".format(dirman_password=dirman_password,
```

Comment:
Instead of `dirman_password=dirman_password` use just `self.master.config.dirman_password` but do not forget to keep values in order.